rescan.go

```diff
@@ -442,6 +442,39 @@ func (rs *rescanState) rescan() error {
 		return err
 	}
 
+	// To ensure that we batch as many filter queries as possible, we also
+	// wait for the header chain to either be current or for it to at least
+	// have caught up with the specified end block.
+	r, ok := chain.(*RescanChainSource)
```

Comment:
Perhaps we can extend the interface to permit these operations w/o needing to type cast?

Reply:
Looks like all we need is the `IsCurrent()` method.

Reply:
good idea :) updated!